### ISSUE TITLE
Keras slow with regularization?

### ISSUE THREAD

**Issue by daquang** [fchollet/keras#863]
I noticed that when I include weight regularization such as L2 weight regularization, it significantly slows down training. Can someone provide an explanation for why that is?

**Comment by fchollet** [comment]
Adding regularization will significantly complicate the computational graph of your model, as you are adding many terms to your loss function. A more complex graph will take longer to run.

A while ago, Keras optimizers were added as part of the optimizer rather than as part of the loss function, which left the computation graph almost unchanged and could run very fast. However it was much less accurate than the current setup. But you could always hack it back in if speed is a big issue for your model.

**Comment by dadhichmohit** [comment]
I have two CNN models, both share same architechture. 'Model A' was trained without (l2)regularization and 'Model B' was trained with (l2)regularization.
on CPU:
-  Model A Prediction time: 10 secs
-  Model B Prediction time :  60 secs 

function used for prediction was 'model.predict()' and rest all conditions like image size, batch_size,number of images were same for both models. Please provide your guidance.
